lib/active_record/connection_adapters/mysql_flexmaster_adapter.rb

```diff
@@ -252,12 +255,14 @@ def initialize_connection(host, port)
               cx.query_options.merge!(:as => :array)
             end
           end
-        rescue Mysql2::Error => e
-          collected_errors << e
-          nil
-        rescue Timeout::Error => e
-          collected_errors << e
-          nil
+        rescue Mysql2::Error, Timeout::Error => e
```

Comment:
wondering what errors other than `Timeout` we want to retry on. it's ok either way, I think...